cloudwatch_metrics_collector/src/metrics_collector.cpp

```diff
@@ -101,10 +115,51 @@ int MetricsCollector::RecordMetrics(
     AWS_LOGSTREAM_DEBUG(__func__, "Recording metric with name=[" << metric_msg->metric_name << "]");
 
     // create a MetricObject with message parameters to batch
-    Aws::CloudWatchMetrics::Utils::MetricObject metric_object {metric_msg->metric_name,
+    Aws::CloudWatchMetrics::Utils::MetricObject metric_object{metric_msg->metric_name,
                                                               metric_msg->value,
                                                               metric_msg->unit,
-                                                              GetMetricDataEpochMillis(*metric_msg),
+                                                              GetMetricDataEpochMillis(metric_msg->time_stamp),
+                                                              dimensions,
+                                                              this->storage_resolution_.load()};
+    bool batched = metric_service_->batchData(metric_object);
+
+    if (!batched) {
+      AWS_LOGSTREAM_ERROR(__func__, "Failed to record metric");
```

Comment:
This should detail which message type failed, otherwise it's the same statement as line 168.

Reply:
Done.